core/src/test/scala/org/apache/spark/deploy/history/FsHistoryProviderSuite.scala

```diff
@@ -335,6 +337,53 @@ class FsHistoryProviderSuite extends FunSuite with BeforeAndAfter with Matchers
     assert(!log2.exists())
   }
 
+  test("Event log copy") {
+    val provider = new FsHistoryProvider(createTestConf())
+    val logs = (1 to 2).map { i =>
+      val log = newLogFile("downloadApp1", Some(s"attempt$i"), inProgress = false)
+      writeFile(log, true, None,
+        SparkListenerApplicationStart(
+          "downloadApp1", Some("downloadApp1"), 5000 * i, "test", Some(s"attempt$i")),
+        SparkListenerApplicationEnd(5001 * i)
+      )
+      log
+    }
+    provider.checkForLogs()
+
+    (1 to 2).foreach { i =>
+      val outDir = Utils.createTempDir()
+      val unzipDir = Utils.createTempDir()
+      try {
+        Utils.chmod700(outDir)
+        Utils.chmod700(unzipDir)
+        val outFile = new File(outDir, s"file$i.zip")
+        val outputStream = new ZipOutputStream(new FileOutputStream(outFile))
+        provider.writeEventLogs("downloadApp1", Some(s"attempt$i"), outputStream)
+        HistoryTestUtils.unzipToDir(new FileInputStream(outFile), unzipDir)
```

Comment:
Not a big deal, but I wonder if this test would be simpler if you read from the zip file directly, instead of having this other method to unzip it.

You could use `new String(ByteStreams.readFully(blah), UTF_8)` to read files from the zip into Strings directly.